src/userCache.ts

```diff
@@ -24,29 +24,29 @@ interface Cache {
 class UserCache {
   cache: Cache = {};
 
-  set(user: User, lifespan: number): void {
-    this.cache[user.email] = {
+  set(user: User, service: string, lifespan: number): void {
+    this.cache[`${user.email}-${service}`] = {
```

Comment:
Maybe just a nitpick, but I would use some character that cannot be used in email for the separation of cache-key. :-) 

E.g. `:`

Reply:
Good point!